### PR TITLE
dcnm_vrf: Updating UT per latest design

### PR DESCRIPTION
**Pull request by nkshrishail**
##### SUMMARY
Updating the UT for DCNM VRF module as the dcnm_vrf module went through a design change.
Ref : https://github.com/mikewiebe-ansible/ansible/pull/11

##### ISSUE TYPE
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME
dcnm_vrf

### CODE REVIEW

File: test/units/modules/network/dcnm/test_dcnm_vrf.py [hunk 1]
```diff
@@ -29,72 +29,280 @@ class TestDcnmVrfModule(TestDcnmModule):
 
     module = dcnm_vrf
 
+    mock_ip_sn = {'10.10.10.224': 'XYZKSJHSMK1',
+                  '10.10.10.225': 'XYZKSJHSMK2'}
+
+    mock_config=\
```
Comment (mikewiebe) [inline test/units/modules/network/dcnm/test_dcnm_vrf.py]: Let's move the mock data into a separate file.

File: test/units/modules/network/dcnm/test_dcnm_vrf.py [hunk 2]
```diff
@@ -29,72 +29,280 @@ class TestDcnmVrfModule(TestDcnmModule):
...
+               },
+              {'ip_address': '10.10.10.225',
+               'vlan_id': '303',
+               'deploy': 'true'
+               }
+          ],
+          'deploy': 'true'
+          }]
+
+    mock_config_incorrect_vrfid = \
```
Comment (mikewiebe) [inline test/units/modules/network/dcnm/test_dcnm_vrf.py]: When I think of mock data it's limited to data we get back from DCNM.  For any data that represents playbook values let's use a naming convention other then `mock_*`.  For example I would switch `mock_config` to something like `playbook_config`.  That we we don't confuse it with data being returned by DCNM.

File: test/units/modules/network/dcnm/test_dcnm_vrf.py [hunk 3]
```diff
@@ -29,72 +29,280 @@ class TestDcnmVrfModule(TestDcnmModule):
...
-            self.run_dcnm_send.side_effect = [deploy_success_resp]
         else:
             pass
 
-    def test_dcnm_vrf_create_new(self):
-        set_module_args(dict(action='create', fabric='test_fabric', vrf_name='test_vrf', vrf_id=919191))
+
+    def test_01dcnm_vrf_merged_new(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.mock_config))
         result = self.execute_module(changed=True, failed=False)
```
Comment (mikewiebe) [inline test/units/modules/network/dcnm/test_dcnm_vrf.py:210]: Looking closer at this test.  Line `267` I believe is redundant with line `268`.  The `self.execute_module` method will automatically assert if `changed` and `failed` keys do not match the values you pass into the function call.  We should be checking the response we expect to be returned from DCNM in this case on line `268`.

For example when I execute your module against our real DCNM instance I see the following in the response.

```json
    "response": {
        "DATA": {
            "successList": [
                {
                    "message": "VRF is successfully created.",
                    "name": "wiebe55-vrf-r1",
                    "status": "Success"
                }
            ]
        },
        "MESSAGE": "OK",
        "METHOD": "POST",
        "REQUEST_PATH": "https://10.122.197.6:443/rest/top-down/bulk-create/vrfs",
        "RETURN_CODE": 200
    }
}
```

We would want to mock this return data and assert on the various bits of info that are returned by the mock object.  This way if the module manipulates this data in any way or introduces a bug we hopefully catch it.  Let me know if you want to discuss in more detail to determine how much of this stuff actually makes sense to test.

**Comment by mikewiebe** [comment]
Just ran the code coverage tool on the `dcnm_vrf` module and its about 69% right now.  Lets chat about how we can add more testing to get more complete coverage.

File: test/units/modules/network/dcnm/test_dcnm_vrf.py (hunk 1 above)
Comment (nkshrishail) [reply test/units/modules/network/dcnm/test_dcnm_vrf.py]: Ok, taken care.

File: test/units/modules/network/dcnm/test_dcnm_vrf.py (hunk 2 above)
Comment (nkshrishail) [reply test/units/modules/network/dcnm/test_dcnm_vrf.py]: ok, taken care

File: test/units/modules/network/dcnm/test_dcnm_vrf.py (hunk 3 above)
Comment (nkshrishail) [reply test/units/modules/network/dcnm/test_dcnm_vrf.py:210]: I think I will include all the fields in the response which are used in the dcnm_vrf module to arrive at a conclusion. Ex: Data, Message and return code.

Comment (nkshrishail) [reply test/units/modules/network/dcnm/test_dcnm_vrf.py:210]: And I never processed the above payload for the vrf create op as my testing was done on API to create single API (which returns blank when it succeeds), I missed to see that response for bulk vrf creation is different. 
But, in any case,  the error checking is primarily based on (or will be in some cases) fields MESSAGE, ERROR and RETURN_CODE, most errors are caught here, only few need the "DATA" to be processed.

File: test/units/modules/network/dcnm/test_dcnm_vrf.py [hunk 4]
```diff
@@ -22,79 +22,274 @@
...
-                             deployment='False', serial_numbers_vlans=[{'947BSAMCAU1':'103'}]))
+        self.assertEqual(result.get('msg'), 'vrf_id for VRF:test_vrf_1 cant be updated to a different value')
+
+    def test_07dcnm_vrf_merged_with_update(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config_update))
+        self.execute_module(changed=True, failed=False)
+
+    def test_08dcnm_vrf_merged_with_update_vlan(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config_update_vlan))
+        self.execute_module(changed=True, failed=False)
```
Comment (mikewiebe) [inline test/units/modules/network/dcnm/test_dcnm_vrf.py]: Maybe I am missing something but I don't see where you are validating any of the return data from the module call in many of these tests?

**Review by mikewiebe — COMMENTED** [review]
Just one comment.

**Pull request merged**